### ISSUE TITLE
Incorrect calculation of Euclidean Norm in measure.compare_nrmse

### ISSUE THREAD

**Issue by MaxSchambach** [scikit-image/scikit-image#2842]
## Description
The function `compare_nrmse()` in the `measure` module uses an incorrect calculation of the Euclidean norm of the input image. The Euclidean or Frobenius norm of a matrix **A** is the square root of the sum of the squared elements of the matrix. The calculation in the `compare_nrmse()` function (line 85) instead calculates the square root of the sum of squared elements divided by the total number of elements.

## Example
As an example, choose
```
im_true = np.array(
            [[0, 1, 1],
             [0, 1, 1],
             [0, 1, 1]])
```

Whose Euclidean / Frobenius norm is evidently `sqrt(6)`.
The used calculation (code line 85)

```
denom = np.sqrt(np.mean((im_true*im_true), dtype=np.float64))
```

yields 0.81649658092772603 which is equivalent to sqrt(2/3)

Line 85 should be changed to

```
denom = np.sqrt(np.sum((im_true*im_true), dtype=np.float64))
```

This however yields an unintuitive normalization for the RMSE, e.g. the NRMSE of to matrices, one consisting of all ones, the other of all zeros, will not be one, but 1/sqrt(# elements)

**Comment by grlee77** [comment]
I think the issue is primarily one of a slightly misleading description in the documentation.  What actually gets computed in that "Euclidean" case is equivalent to:    `np.linalg.norm(im_true - im_test) / np.linalg.norm(im_true)`.  Assuming that is the desired result, then the code is correct as is.  Effectively the 1/N due to `np.mean()` in the denominator here cancels out the 1/N due to `np.mean()` in the numerator.  That variant is not mentioned on the wikipedia page that was referenced, but I have found this form fairly common in the medical imaging literature.  Here are two specific examples of publications using that form :
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3545284/    (see Eq. 25)
https://arxiv.org/abs/1604.04986    (see Eq. 7)

Have you found a convention where the sqrt(N) is kept within the numerator, but not the denominator?  If that is also fairly common, it might be worth making both variants available.

In either case, the equation used in this "Euclidean" case should be stated explicitly in the docstring for clarity.

**Comment by MaxSchambach** [comment]
Yes, you are right.
I think stating what you have explained in the documentation, in particular the formula

NRMSE = || (im_true - im_test) || / ||im_true||

will resolve all confusion with the name "Euclidean". I have not found a convention where sqrt(N) is kept, as this also yields quite unintuitive normalization.

**Comment by stefanv** [comment]
If you are agreed that this is a reasonable formulation, then please make a PR to update the docs; I'd be happy to review it.

**Comment by MaxSchambach** [comment]
Thanks, I open a corresponding pull request #2855

**Comment by stefanv** [comment]
Closed by #2855
